src/main/java/uk/gov/hmcts/reform/authorisation/filters/ServiceAuthFilter.java

```diff
@@ -0,0 +1,68 @@
+package uk.gov.hmcts.reform.authorisation.filters;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.filter.OncePerRequestFilter;
+import uk.gov.hmcts.reform.authorisation.exceptions.InvalidTokenException;
+import uk.gov.hmcts.reform.authorisation.exceptions.ServiceException;
+import uk.gov.hmcts.reform.authorisation.validators.AuthTokenValidator;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Collectors;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class ServiceAuthFilter extends OncePerRequestFilter {
+
+    public static final String AUTHORISATION = "ServiceAuthorization";
+
+    private static final Logger LOG = LoggerFactory.getLogger(ServiceAuthFilter.class);
+
+    private final List<String> authorisedServices;
+
+    private final AuthTokenValidator authTokenValidator;
+
+    public ServiceAuthFilter(AuthTokenValidator authTokenValidator, List<String> authorisedServices) {
+
+        this.authTokenValidator = authTokenValidator;
+        if (authorisedServices.isEmpty()) {
```

Comment:
could this be null?
```suggestion
        if (authorisedServices == null || authorisedServices.isEmpty()) {
```

Reply:
I was worried that I will be checking null within a list but you are right no harm even if it did.